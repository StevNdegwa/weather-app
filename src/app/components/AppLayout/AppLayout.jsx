import { Box, Fade } from "@mui/material";

export default function AppLayout({ children }) {
  return (
    <Fade in={true}>
      <Box
        style={{
          width: "100vw",
          height: "100vh",
          backgroundImage:
            "linear-gradient(#5BBCF2, #90CBEC, #AED2E5, #DAE3E8, #F4F4F4)",
        }}
      >
        {children}
      </Box>
    </Fade>
  );
}
