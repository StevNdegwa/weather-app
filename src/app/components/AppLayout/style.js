import { Box, styled } from "@mui/material";

export const AppLayoutWrapper = styled(Box)`
width: 100vw;
height: 100vh;
position: relative;
overflow: auto;
background-image: linear-gradient(#5BBCF2, #90CBEC, #AED2E5, #DAE3E8, #F4F4F4);
`;

export const WeatherIconContainer = styled(Box)`
position: absolute;
top: 0;
right: 0;
`;