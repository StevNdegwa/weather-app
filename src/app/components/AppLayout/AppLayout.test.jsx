import { shallow } from "enzyme";
import { Box, Fade } from "@mui/material";
import AppLayout from "./AppLayout";

describe("test <AppLayout/> component", () => {
  let wrapper;

  beforeEach(
    () =>
      (wrapper = shallow(
        <AppLayout>
          <span />
        </AppLayout>
      ))
  );

  test("should contain a Box component", () => {
    expect(wrapper.find(Box)).toHaveLength(1);
  });

  test("should contain the Fade component", () => {
    expect(wrapper.find(Fade)).toHaveLength(1);
  })

  test("should render the component children", () => {
    expect(wrapper.childAt(0).childAt(0).type()).toBe("span");
  });
});
