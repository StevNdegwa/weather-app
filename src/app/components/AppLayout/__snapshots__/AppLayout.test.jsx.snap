// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <AppLayout/> component snapshot testing 1`] = `"<div class=\\"MuiBox-root css-1xmafg8\\"><span></span><div class=\\"MuiBox-root css-1y08cah\\"><lottie-player src=\\"https://assets3.lottiefiles.com/packages/lf20_xx4r8tdi.json\\" background=\\"transparent\\" speed=\\"1\\" style=\\"width: 100px; height: 100px;\\" loop=\\"true\\" autoplay=\\"true\\"></lottie-player></div></div>"`;
